tests/src/unit-class_parser.cpp

```diff
@@ -1644,6 +1644,14 @@ TEST_CASE("parser class")
 
         SECTION("SAX parser")
         {
+            SECTION("null sax handler")
+            {
+
+                std::string s = "some_string";
+                SaxCountdown* p = nullptr;
+                nlohmann::json j;
+                CHECK_THROWS_AS(j.sax_parse(s, p), json::other_error);
```

Comment:
Please use `CHECK_THROWS_WITH_AS` like above to check both the type of the exception and the error message.